build-arch-gce

```diff
@@ -154,7 +144,7 @@ arch-chroot -- "$mount_dir" /bin/bash -s <<-'EOS'
 	gawk -i assert -i inplace '
 		/^MODULES=/ { $0 = "MODULES=(virtio_pci virtio_scsi sd_mod ext4)"; ++f1 }
 		/^BINARIES=/ { $0 = "BINARIES=(fsck fsck.ext4)"; ++f2 }
-		/^HOOKS=/ { $0 = "HOOKS=(base modconf)"; ++f3 }
+		/^HOOKS=/ { $0 = "HOOKS=(base systemd autodetect modconf block filesystems fsck)"; ++f3 }
```

Comment:
autodetect only works if you are guaranteed the image will be used on the same HW where this script runs. I don't think it's a good idea for these images. The required modules for boot on GCE are spelled out manually at line 145, so I think that should be enough.

Another note: using systemd for initramfs here is completely independent from systemd-boot. You could do the same on grub. It makes initramfs bigger compared to now though.

Reply:
Got it.